doc/changelog.md

```diff
@@ -143,6 +143,24 @@
       return qml.expval(qml.Hamiltonian(coeffs, obs))
   ```
 
+* Add support for nested Hamiltonian observables.

```

Comment:
I think this PR (and 248 just above) would fit better in the *Improvements* section than in `New features* section. I know it's not always clear which is which but here the impact is fairly small compared to something like the Enzyme support.